test/e2e/serial/tests/workload_placement.go

```diff
@@ -413,6 +415,305 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
+
+			rl0 := e2ereslist.FromGuaranteedPod(updatedPod0)
+			klog.Infof("post-create pod resource list: spec=[%s] updated=[%s]", e2ereslist.ToString(e2ereslist.FromContainers(podSpec.Containers)), e2ereslist.ToString(rl0))
+
+			nrtInitial0, err := e2enrt.FindFromList(nrtInitialList.Items, updatedPod0.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			nrtPostCreate0, err := e2enrt.FindFromList(nrtPostCreateDeploymentList.Items, updatedPod0.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedPod1 := pods[1]
+			By(fmt.Sprintf("checking the pod landed on the target node %q vs %q", updatedPod1.Spec.NodeName, targetNodeName))
+			Expect(updatedPod1.Spec.NodeName).To(Equal(targetNodeName),
+				"node landed on %q instead of on %v", updatedPod1.Spec.NodeName, targetNodeName)
+
+			By(fmt.Sprintf("checking the pod was scheduled with the topology aware scheduler %q", serialconfig.Config.SchedulerName))
+			schedOK, err = nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod1.Namespace, updatedPod1.Name, serialconfig.Config.SchedulerName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod1.Namespace, updatedPod1.Name, serialconfig.Config.SchedulerName)
+
+			rl1 := e2ereslist.FromGuaranteedPod(updatedPod1)
+			klog.Infof("post-create pod resource list: spec=[%s] updated=[%s]", e2ereslist.ToString(e2ereslist.FromContainers(podSpec.Containers)), e2ereslist.ToString(rl1))
+
+			nrtInitial1, err := e2enrt.FindFromList(nrtInitialList.Items, updatedPod1.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			nrtPostCreate1, err := e2enrt.FindFromList(nrtPostCreateDeploymentList.Items, updatedPod1.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			var checkConsumedRes checkConsumedResFunc
+			if nrtInitial0.TopologyPolicies[0] == string(nrtv1alpha1.SingleNUMANodePodLevel) {
+				// TODO: this is only partially correct. We should check with NUMA zone granularity (not with NODE granularity)
+				checkConsumedRes = e2enrt.CheckZoneConsumedResourcesAtLeast
+			} else {
+				checkConsumedRes = e2enrt.CheckNodeConsumedResourcesAtLeast
+			}
+
+			By(fmt.Sprintf("checking post-create NRT after pod: %q for target node %q updated correctly", updatedPod0.Name, targetNodeName))
+			dataBefore0, err := yaml.Marshal(nrtInitial0)
+			Expect(err).ToNot(HaveOccurred())
+			dataAfter0, err := yaml.Marshal(nrtPostCreate0)
+			Expect(err).ToNot(HaveOccurred())
+			match0, err := checkConsumedRes(*nrtInitial0, *nrtPostCreate0, rl0)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(match0).ToNot(BeEmpty(), "inconsistent accounting: no resources consumed by the running pod,\nNRT before test's pod: %s \nNRT after: %s \n total required resources: %s", dataBefore0, dataAfter0, e2ereslist.ToString(rl0))
+
+			if nrtInitial1.TopologyPolicies[0] == string(nrtv1alpha1.SingleNUMANodePodLevel) {
```

Comment:
since both replicas will be on the same node, I am not confident this (& prev nrt check) will pass, because updatedPod1.Spec.NodeName & updatedPod0.Spec.NodeName points to the same node, so they'll have same nrts post deployment creation

Reply:
Yep, that is correct. I will fix. Rather than checking after each replica we need to check once for the deployment as a whole.